workspaces/src/types/mod.rs

```diff
@@ -28,25 +28,79 @@ pub type Balance = u128;
 /// Height of a specific block
 pub type BlockHeight = u64;
 
+/// Key types supported for either a [`SecretKey`] or [`PublicKey`]
+#[derive(Debug, Copy, Clone, Serialize, Deserialize)]
+#[non_exhaustive]
+pub enum KeyType {
+    ED25519,
+    SECP256K1,
+}
+
+impl From<KeyType> for near_crypto::KeyType {
+    fn from(key_type: KeyType) -> Self {
+        match key_type {
+            KeyType::ED25519 => Self::ED25519,
+            KeyType::SECP256K1 => Self::SECP256K1,
+        }
+    }
+}
+
+impl From<near_crypto::KeyType> for KeyType {
+    fn from(key_type: near_crypto::KeyType) -> Self {
+        match key_type {
+            near_crypto::KeyType::ED25519 => Self::ED25519,
+            near_crypto::KeyType::SECP256K1 => Self::SECP256K1,
+        }
+    }
+}
+
 impl From<PublicKey> for near_crypto::PublicKey {
     fn from(pk: PublicKey) -> Self {
         pk.0
     }
 }
 
-#[derive(Clone, Serialize, Deserialize)]
+/// Public key of an account on chain. Usually created along with a [`SecretKey`]
+/// to form a keypair associated to the account.
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct PublicKey(pub(crate) near_crypto::PublicKey);
 
-#[derive(Clone, Serialize, Deserialize)]
+impl PublicKey {
+    pub fn key_type(&self) -> KeyType {
+        self.0.key_type().into()
+    }
+}
+
+/// Secret key of an account on chain. Usually created along with a [`PublicKey`]
+/// to form a keypair associated to the account. To generate a new keypair, use
+/// one of the creation methods found here, such as [`SecretKey::from_seed`]
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub struct SecretKey(near_crypto::SecretKey);
 
 impl SecretKey {
+    pub fn key_type(&self) -> KeyType {
+        self.0.key_type().into()
+    }
+
     pub fn public_key(&self) -> PublicKey {
         PublicKey(self.0.public_key())
     }
 
     pub fn from_seed(key_type: KeyType, seed: &str) -> Self {
-        Self(near_crypto::SecretKey::from_seed(key_type, seed))
+        Self(near_crypto::SecretKey::from_seed(key_type.into(), seed))
+    }
+
+    pub fn from_random(key_type: KeyType) -> Self {
+        Self(near_crypto::SecretKey::from_random(key_type.into()))
+    }
+}
+
+impl std::str::FromStr for SecretKey {
+    // FIXME: deferred specific error impl to https://github.com/near/workspaces-rs/issues/100
+    type Err = anyhow::Error;
+
+    fn from_str(value: &str) -> anyhow::Result<Self> {
+        Ok(Self(near_crypto::SecretKey::from_str(value)?))
```

Comment:
noting that we will have to be careful on updating this dep that the semantics don't change. Not sure how stable this API is